examples/gno.land/r/demo/tamagotchi/tamagotchi.gno

```diff
@@ -0,0 +1,185 @@
+package tamagotchi
+
+// this file could be moved to p/demo/tamagotchi.
+
+import (
+	"time"
+
+	"gno.land/p/demo/ufmt"
+)
+
+// Tamagotchi structure
+type Tamagotchi struct {
+	name        string
+	hunger      uint
+	happiness   uint
+	health      uint
+	age         uint
+	maxAge      uint
+	sleepy      uint
+	created     time.Time
+	lastUpdated time.Time
+}
+
+func New(name string) *Tamagotchi {
+	now := time.Now()
+	return &Tamagotchi{
+		name:        name,
+		hunger:      50,
+		happiness:   50,
+		health:      50,
+		maxAge:      100,
+		lastUpdated: now,
+		created:     now,
+	}
+}
+
+func (t *Tamagotchi) Name() string {
+	t.update()
+	return t.name
+}
+
+func (t *Tamagotchi) Hunger() uint {
+	t.update()
+	return t.hunger
+}
+
+func (t *Tamagotchi) Happiness() uint {
+	t.update()
+	return t.happiness
+}
+
+func (t *Tamagotchi) Health() uint {
+	t.update()
+	return t.health
+}
+
+func (t *Tamagotchi) Age() uint {
+	t.update()
+	return t.age
+}
+
+func (t *Tamagotchi) Sleepy() uint {
+	t.update()
+	return t.sleepy
+}
+
+// Feed method for Tamagotchi
+func (t *Tamagotchi) Feed() {
+	t.update()
+
+	if t.health == 0 { // dead
+		return
+	}
+
+	t.hunger -= 10
+	if t.hunger < 0 {
+		t.hunger = 0
+	}
+}
+
+// Play method for Tamagotchi
+func (t *Tamagotchi) Play() {
+	t.update()
+
+	if t.health == 0 { // dead
+		return
+	}
+
+	t.happiness += 10
+	if t.happiness > 100 {
+		t.happiness = 100
+	}
+}
+
+// Heal method for Tamagotchi
+func (t *Tamagotchi) Heal() {
+	t.update()
+
+	if t.health == 0 { // dead
+		return
+	}
+
+	t.health += 10
+	if t.health > 100 {
+		t.health = 100
+	}
+}
+
+// Update applies changes based on the duration since the last update
+func (t *Tamagotchi) update() {
+	now := time.Now()
+	if t.lastUpdated == now {
+		return
+	}
+
+	duration := now.Sub(t.lastUpdated)
+	elapsedMins := uint(duration.Minutes())
+
+	t.hunger += elapsedMins
+	if t.hunger > 100 {
+		t.hunger = 100
+	}
```

Comment:
simplify these to a helper function `func bound(n, min, max int) int`?

Reply:
https://github.com/gnolang/gno/pull/1359/commits/ad8a1e5266d95cf4669ed208b814ac74f62068ce